pdal/SpatialReference.cpp

```diff
@@ -53,22 +53,13 @@
 namespace
 {
 
-struct OGRDeleter
```

Comment:
Was there a reason for removing this deleter? All of the SRS handling code expects these to clean up after themselves and is going to leak unless I'm misunderstanding something.

Reply:
The cleanup isn't going away. I just remove the need for the *custom*  deleter by letting `unique_ptr<OGRSpatialReference>` fully embrace the C++ nature of the managed pointer and use the *default* deleter. 

See also the following lines: I changed construction to `new OGRSpatialReference(...)` instead of `static_cast<OGRSpatialReference*>(OSRNewSpatialReference(...))`.

I did not change the managed type.

Reply:
> I'm not sure the history of why there's all specialized deleter functions that don't do anything specialized.

Because the interface predates working smart pointers? Or maybe there was specialized behavior in them at one time 🤷